internal/sbi/nsselection/routers.go

```diff
@@ -38,13 +42,24 @@ type Routes []Route
 // NewRouter returns a new router.
 func NewRouter() *gin.Engine {
 	router := logger_util.NewGinWithLogrus(logger.GinLog)
+	routerAuthorizationCheck := util.NewRouterAuthorizationCheck(serviceName)
+
+	router.Use(func(c *gin.Context) {
+		routerAuthorizationCheck.Check(c, nssf_context.GetSelf())
+	})
+
 	AddService(router)
 	return router
 }
 
 func AddService(engine *gin.Engine) *gin.RouterGroup {
 	group := engine.Group(factory.NssfNsselectResUriPrefix)
 
+	routerAuthorizationCheck := util.NewRouterAuthorizationCheck(serviceName)
```

Comment:
Do we need add the middleware `NewRouterAuthorizationCheck` in both `NewRouter()` and `AddService()`?

Reply:
We don't. Since there's no usage on `NewRouter()`, I have removed the middleware in `NewRouter()` and keep the one in `AddService()`.